plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -798,12 +807,40 @@ private List<ColumnMetadata> getTableColumnMetadata(MetadataEntry metadataEntry,
     @Override
     public List<SchemaTableName> listTables(ConnectorSession session, Optional<String> schemaName)
     {
-        return schemaName.map(Collections::singletonList)
+        return streamTables(session, schemaName)
+                .map(TableInfo::tableName)
+                .collect(toImmutableList());
+    }
+
+    @Override
+    public Map<SchemaTableName, RelationType> getRelationTypes(ConnectorSession session, Optional<String> schemaName)
+    {
+        return streamTables(session, schemaName)
+                .collect(toImmutableMap(TableInfo::tableName, this::resolveRelationType, (ignore, second) -> second));
+    }
+
+    private Stream<TableInfo> streamTables(ConnectorSession session, Optional<String> optionalSchemaName)
+    {
+        List<Callable<List<TableInfo>>> tasks = optionalSchemaName.map(Collections::singletonList)
                 .orElseGet(() -> listSchemaNames(session))
                 .stream()
-                .flatMap(schema -> metastore.getAllTables(schema).stream()
-                        .map(table -> new SchemaTableName(schema, table)))
+                .map(schemaName -> (Callable<List<TableInfo>>) () -> metastore.getAllTables(schemaName))
                 .collect(toImmutableList());
+        try {
+            return processWithAdditionalThreads(tasks, metadataFetchingExecutor).stream()
+                    .flatMap(Collection::stream);
+        }
+        catch (ExecutionException e) {
+            throw new RuntimeException(e.getCause());
+        }
+    }
+
+    private RelationType resolveRelationType(TableInfo tableInfo)
```

Comment:
@piotrrzysko I would drop the question from the commit message.
The question was:

> Is this resolution necessary? Could we instead use the
> existing mapping between ExtendedRelationType and RelationType that's
> already encapsulated in RelationType?

I guess the question here is, is `OTHER_VIEW` or `OTHER_MATERIALIZED_VIEW` possible in delta-lake? `TRINO_MATERIALIZED_VIEW` is not.
@raunaqmorarka @dain Do you know?

Reply:
btw it is ok for me to have the `resolveRelationType` here to keep the existing functionality